pkg/tfvars/internal/cache/cache.go

```diff
@@ -243,5 +244,16 @@ func DownloadImageFile(baseURL string) (string, error) {
 	}
 	u.location = *parsedURL
 
-	return u.download(imageDataType)
+	for i := 0; i < 3; i++ {
```

Comment:
I understand that this part was already duplicated, it would be nice to find a way at least to avoid increasing it (cc @bfournie that may have some better insight for that)

Reply:
I'm open to suggestions. Could we e.g. move this under rhcos module?

Reply:
Sorry I was on PTO when the cc above was added and did not notice it.

The reason the function was duplicated, as you probably know, was because its in an `internal` dir and can't be accessed from `pkg/asset/agent/`.
I think moving it to `pkg/rhcos` may be an option as that's already included by files both in `pkg/tfvars/` and in `pkg/asset/agent/`